modules/navier_stokes/src/auxkernels/kEpsilonViscosityAux.C

```diff
@@ -166,56 +124,83 @@ kEpsilonViscosityAux::computeValue()
     // Compute the velocity and direction of the velocity component that is parallel to the wall
     const ADReal parallel_speed = (velocity - velocity * loc_normal * loc_normal).norm();
 
-    ADReal y_plus, u_tau;
-    if (_non_equilibrium_treatment)
+    // Switch for determining the near wall quantities
+    // wall_treatment can be: "eq_newton eq_incremental eq_linearized neq"
+    ADReal u_tau;
+    ADReal y_plus;
+    ADReal mut_log; // turbulent log-layer viscosity
+    ADReal mu_wall; // total wall viscosity to obtain the shear stress at the wall
+
```

Comment:
The whole method blow needs to be revised to optimize memory usage and efficiency. It should read as follows:

compute y_plus regarding the wall treatment method used;
if (y_plus <= 5.0)
  return 0.0
else if (y_plus >= 30.0)
  // do all the things that you are doing for the log-layer treatment
else
 // do the methd for the buffer layer

Reply:
The way I see this implemented is as follows:

if (wall_treatment = 'neq, eq_new, eq_incr, eq_lin')
     compute yplus 
     compute u_tau

if (yplus<5)
     0
else if (yplus>30)
     if (neq or eq_lin)
          mu_w = mu.(y+ kappa/log... - 1)
          mut_log=muw-mu
     else
          mu_w = utau^2 rho ....
          mut_log=muw-mu
else 

you save operations for the laminar case, but introduces a new if statement for the log layer cases that might make it more complicated to read. What do you think?